WhatsApp/app/src/main/java/com/example/whatsapp/ui/fragments/contactsselect/ContactsSelectAdapter.kt

```diff
@@ -0,0 +1,180 @@
+import androidx.recyclerview.widget.RecyclerView
+import com.example.firestorerepository.datatypes.Contact
+import com.example.whatsapp.R
+import com.example.whatsapp.databinding.ItemContactBinding
+
+private const val LOGTAG = "ContactsSelectAdapterTest"
+private const val ADDEDED_HEADERS = 2
+private const val ITEM_VIEW_TYPE_NEW_GROUP = 0
+private const val ITEM_VIEW_TYPE_NEW_CONTACT = 1
+private const val ITEM_VIEW_TYPE_CONTACTS = ADDEDED_HEADERS
+
+class ContactsSelectAdapter :
+    ListAdapter<Contact, RecyclerView.ViewHolder>(DiffCallback()) {
+
+    var tracker: SelectionTracker<Long>? = null
+
+    init {
+        setHasStableIds(true)
+    }
+
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): RecyclerView.ViewHolder {
+
+        return when (viewType) {
+            ITEM_VIEW_TYPE_NEW_GROUP -> NewGroupViewHolder.from(parent)
+            ITEM_VIEW_TYPE_NEW_CONTACT -> NewContactViewHolder.from(parent)
+            ITEM_VIEW_TYPE_CONTACTS -> ContactsViewHolders.from(parent)
+            else -> throw ClassCastException("Unknown viewType $viewType")
+        }
+    }
+
+    class ContactsViewHolders(val binding: ItemContactBinding) : RecyclerView.ViewHolder(binding.root) {
+
+        private val tvName = binding.tvName
+        private val tvDescription = binding.tvDescription
+        private val lavSelected = binding.lavSelected
+
+        fun bind(contact: Contact, isSelected : Boolean) {
+            lavSelected.visibility = if(isSelected) View.VISIBLE else View.GONE
+            itemView.isSelected = isSelected
+            tvName.text = "Oskar $itemId"
+            tvDescription.text = "example :)"
+            Log.i(LOGTAG, "ITEM ID: $itemId ITEM NUM: $contact (ContactsVH)")
+        }
+
+        fun getItemDetails(): ItemDetailsLookup.ItemDetails<Long> =
+            object : ItemDetailsLookup.ItemDetails<Long>() {
+                override fun getPosition(): Int = absoluteAdapterPosition
+                override fun getSelectionKey(): Long = itemId
+            }
+
+        companion object {
+            fun from(parent: ViewGroup): ContactsViewHolders {
+                val layoutInflater = LayoutInflater.from(parent.context)
+                val view = ItemContactBinding.inflate(layoutInflater, parent, false)
+                return ContactsViewHolders(view)
+            }
+        }
+    }
+
+    class NewGroupViewHolder(view: View) : RecyclerView.ViewHolder(view) {
+        init {
+            Log.i(LOGTAG, "ITEM ID: $itemId (NewGroupVH)")
+        }
+        companion object {
+            fun from(parent: ViewGroup): NewGroupViewHolder {
+                val layoutInflater = LayoutInflater.from(parent.context)
+                val view = layoutInflater.inflate(R.layout.item_new_group, parent, false)
+                return NewGroupViewHolder(view)
+            }
+        }
+    }
+
+    class NewContactViewHolder(view: View) : RecyclerView.ViewHolder(view) {
```

Comment:
`    class NewGroupViewHolder(val binding: ItemNewContactBinding) : RecyclerView.ViewHolder(binding.root) {
`

Replace it with view binding so you can avoid using findViewByid() :)